packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx

```diff
@@ -72,20 +90,41 @@ const PieChartContextMenu: FC<PieChartContextMenuProps> = ({
     };
 
     const handleOpenUnderlyingDataModal = () => {
-        // TODO: implement this
-        // openUnderlyingDataModal({
-        //     item,
-        //     value,
-        //     fieldValues: underlyingFieldValues,
-        // });
-        // track({
-        //     name: EventName.VIEW_UNDERLYING_DATA_CLICKED,
-        //     properties: {
-        //         organizationId: user?.data?.organizationUuid,
-        //         userId: user?.data?.userUuid,
-        //         projectId: projectUuid,
-        //     },
-        // });
+        if (!pieChartConfig.selectedMetric) return;
+
+        openUnderlyingDataModal({
+            item: pieChartConfig.selectedMetric,
+            value,
+            fieldValues: {},
+            dashboardFilters: dashboardFiltersThatApplyToChart,
+            pivotReference: {
+                field: getItemId(pieChartConfig.selectedMetric),
+                pivotValues: groupDimensions
```

Comment:
Is the `pivotReference` here just the dimension that splits the pie chart up? I'm not sure I completely follow the logic here. Is this something that we can just get from `pieChartConfig` similarly to how you do `pieChartConfig.selectedMetric` ?